_posts/2016-08-09-the-tech-behind-live-auction-integration.markdown

```diff
@@ -0,0 +1,57 @@
+author: alan
+categories: [engineering, architecture, auctions, javascript, react, redux, scala, akka, swift, launch, lifecycle]
+---
+
+In late June, the [Artsy auctions](https://www.artsy.net/auctions/) team launched our Live Auction Integration (LAI) product. To summarize its purpose, it allows people to participate online in live sales held at auction houses [that partner with Artsy](https://www.artsy.net/auction-partnerships). It was a big project, begun in December, involving both brand new software and extensive integration work with the rest of our systems. This is the first in what will be a series of blog posts discussing the engineering work we did to get a complex product from inception to launch in such a brief time window, with a go-live deadline set in stone weeks in advance. In this, I’ll dig into what we shipped on a high level, as well as some of the overarching technical decisions.
+
+<!-- more -->
+
+LAI raised challenges that were novel to our engineering team. The product is a real-time experience from the perspectives of the bidder and the auction house, and producing that experience requires a complex human + computer system. There are two main flows of information: from auction house to bidder, and vice versa. These are mediated by our systems and staff as follows:
+
+- As bids occur in the auction house sale room, an Artsy operator working on-site inputs that activity into a web interface so that online participants can keep track of what’s happening.
+- As online participants place bids, our system records those as “prospective bids”, and an Artsy bidding clerk on site at the auction house bids on their behalf in the auction house. As those bids are recognized, they are reflected back to all participants, through the prior flow.
+
+This needs to happen in a tight loop to allow online bidders to be competitive with those in the room and on the phone, and the architecture and UX of the LAI product were optimized for that result. In addition, because we are integrating with live events outside of our direct control, there are many, many ways things can deviate from this idealized flow. We had to carefully account for these situations.
+
+Where possible, we wanted to leverage our existing auctions technology. But in some places, there were opportunities to upgrade that technology and other places where new approaches were necessary. Meanwhile, we were running the busiest Artsy auction season to date, and we had to ensure that we weren’t disrupting our existing stack. Below, I discuss the pieces of the end-to-end product.
+
+# The Live user experience
+
+When the live auction actually begins, participants and Artsy staff interact with the system with front-end software developed from scratch. Web users (desktop and mobile) and staff use a new, dedicated Artsy Live web application, which is implemented in a project we call Prediction. iOS Artsy App users can also participate with newly developed UX within that app.
```

Comment:
Ah yes, meant to link that. Will do!

Yeah, I want to discuss as a product+engineering team open source plans for Prediction and even Causality -- whether and when we want to do so in both cases.